src/game/main.cpp

```diff
@@ -150,24 +154,59 @@ void Check_Windowed(int argc, char *argv[])
         if (strcasecmp(argv[i], "-xpos") == 0) {
             ++i;
             g_xPos = atoi(argv[i]);
-            g_xPos = std::clamp(g_xPos, 0, (int)(Res.right - 800)); // Prevent negative values
         }
 
         if (strcasecmp(argv[i], "-ypos") == 0) {
             ++i;
             g_yPos = atoi(argv[i]);
-            g_yPos = std::clamp(g_yPos, 0, (int)(Res.bottom - 600)); // Prevent negative values
         }
     }
+#ifdef PLATFORM_WINDOWS
+    RECT Res;
+    GetWindowRect(GetDesktopWindow(), &Res);
+    g_xPos = std::clamp(g_xPos, 0, (int)(Res.right - WIN_WIDTH)); // Prevent negative values
+    g_yPos = std::clamp(g_yPos, 0, (int)(Res.bottom - WIN_HEIGHT)); // Prevent negative values
 #endif
 }
 
 void Create_Window()
```

Comment:
May I suggest to leave `Create_Window` as is and simply create a new function `Create_Window_SDL2` that is called instead of this one? This way we do not need to touch this function at all and leave it in pristine state.

Reply:
Yes, that would maybe help readability. Will do so